Source/DafnyCore/DafnyOptions.cs

```diff
@@ -34,10 +35,41 @@ public enum QuantifierSyntaxOptions {
     Version4,
   }
 
+  public interface IDafnyOutputWriter {
+    void Debug(string message);
+
+    [StringFormatMethod(nameof(message))]
+    void Debug(string message, params object[] arguments) {
+      Debug(string.Format(message, arguments));
+    }
+
+    void Exception(string message);
+    Task Status(string message);
+    Task Raw(string message);
```

Comment:
Why are some of the interface methods sync and others async? (Also, please add doc comments for these.)

Reply:
I didn't have a principled approach for that difference, but `Debug` and `Exception` are often called from non-async contexts, so I didn't make those async.